Managers/PlayerManager.gd

```diff
@@ -4,13 +4,42 @@ extends Node
 ## Since we only have one player, you can easily get the player from anywhere by calling
 ## PlayerManager.player.
 
+signal on_player_initialized
 
 var player: Player
+var player_position: Vector2i
 
-signal on_player_initialized
+
+func _ready() -> void:
+	player = null
+	MapManager.map_initialized.connect(_set_starting_position)
+	
+
+func _set_starting_position() -> void:
+	player_position = _get_starting_position()
 
 
 func set_player(in_player: Player) -> void:
 	player = in_player
 	if player != null:
 		on_player_initialized.emit()
+		
+
+func _get_starting_position() -> Vector2i:
+	var current_map_width_array: Array[int] = MapManager.map_width_array
+	var starting_position: Vector2i = Vector2i(0,0)
+	
+	var padding_size_first_floor: int = (current_map_width_array.max() - current_map_width_array[0])/2
+	
+	starting_position.x = padding_size_first_floor
+	
+	return starting_position
```

Comment:
If I understood this correctly, this returns back the starting position as the mid point of the first floor. I believe design team mentioned that there can be a first floor where we can select where we can start, should we address that in this Issue or make a new one for it?

Reply:
This returns the first room of the floor (so the most left room)
Indeed player could select their room at start. I thought of having a special state were you can have access to all the rooms of the first floor.
I can do that with a boolean like `selecting_first_floor`, with an associated room selection function that returns all the rooms of the first floor

Reply:
Ok, just wanted to make sure we have that in mind. I'll resolve this for now, but if we add this new state in this PR, just go ahead and reopen this comment and I can take another look.

Reply:
i'll do it now, it's not too hard to do and it makes sense to do it now